### PR TITLE
Refactor/get fuel type/separate by make

### PR DESCRIPTION
**Pull request by LuisFros**
[sc-73305]

### CODE REVIEW

File: src/resolvers/get-fuel-type.js [hunk 1]
```diff
@@ -2,24 +2,66 @@ const Make = require('../constants/make')
...
+        return FuelType.ELECTRIC
+      }
+      break
+    default:
+      break
   }
   // Some of the following regexes uses negative look-behinds to account for
   // malicious input that will cause them to perform poorly.
   //
   // https://codeql.github.com/codeql-query-help/javascript/js-polynomial-redos/
```
Comment (mex) [inline src/resolvers/get-fuel-type.js:61]: This comment belongs with the regex `/((?<!(a3))a3.+e-tron|(?<!(q7))q7.+e-tron)/i`.

File: src/resolvers/get-fuel-type.js [hunk 2]
```diff
@@ -2,24 +2,66 @@ const Make = require('../constants/make')
...
-  if (
-    description.match(/(gte|(?<!(a3))a3.+e-tron|(?<!(q7))q7.+e-tron|tfsi ?e)/i) ||
-    (description.match(/hybrid/i) && !description.match(/mild hybrid/i))
-  ) {
+  if (description.match(/(tfsi ?e)/i) || (description.match(/hybrid/i) && !description.match(/mild hybrid/i))) {
```
Comment (mex) [inline src/resolvers/get-fuel-type.js:62]: `description.match(/(tfsi ?e)/i)` is also Audi specific.

File: src/resolvers/get-fuel-type.js [hunk 3]
```diff
@@ -2,24 +2,66 @@ const Make = require('../constants/make')
...
+      }
+      if (description.match(/e-tron/i)) {
+        return FuelType.ELECTRIC
+      }
+      break
+    case Make.VOLKSWAGEN:
+      if (description.match(/gte/i)) {
+        return FuelType.HYBRID
+      }
+      if (description.match(/(e-golf|e-up!|e-up( |$)|id\.[3-5]|e-crafter)/i)) {
```
Comment (mex) [inline src/resolvers/get-fuel-type.js:38]: `e-crafter` is also a model for MAN, I believe, so this probably shouldn't be VW specific.

File: src/resolvers/get-fuel-type.js [hunk 4]
```diff
@@ -2,24 +2,66 @@ const Make = require('../constants/make')
...
+ * @param {Object} vehicle
+ * @param {string} vehicle.vin
+ * @param {string} vehicle.name
+ * @returns {Fuels|null}
+ */
+const getFuelTypeFromVehicleInfo = vehicle => {
+  const description = vehicle.name
+  if (description === 'SKODA OCTAVIA A8 COMBI Style 1ED 1,0 eTSI 110') {
+    console.log('')
+  }
```
Comment (mex) [inline src/resolvers/get-fuel-type.js:20]: This probably shouldn't be here.

File: src/resolvers/get-fuel-type.js [hunk 5]
```diff
@@ -2,24 +2,66 @@ const Make = require('../constants/make')
...
+      }
+      if (description.match(/(e-golf|e-up!|e-up( |$)|id\.[3-5]|e-crafter)/i)) {
+        return FuelType.ELECTRIC
+      }
+      break
+    case Make.SKODA:
+      if (description.match(/citigoe/i)) {
+        return FuelType.ELECTRIC
+      }
+      if (description.match(/ iV /i)) {
```
Comment (mex) [inline src/resolvers/get-fuel-type.js:46]: Might as well keep it lowercase as you ignore casing:

```suggestion
      if (description.match(/ iv /i)) {
```

File: src/resolvers/get-fuel-type.js (hunk 4 above)
Comment (LuisFros) [reply src/resolvers/get-fuel-type.js:20]: 🙈

**Review by mex — APPROVED** [review]

**Pull request merged**